Multinode xena

jackhodgkiss opened the pull request.

jackhodgkiss committed:
feat:  copy `ci-multinode` from PR `#246`

jackhodgkiss committed:
feat: copy `ansible` from PR `#246`

jackhodgkiss committed:
revert: configure-aio-resources.yml to `da05599`

jackhodgkiss committed:
feat: actually copy contents of `ci-multinode`

For some reason the previous commit does not contain the important
changes to the configuration. This commit fixes that.

jackhodgkiss committed:
feat: include `.automation.conf` from PR `#246`

jackhodgkiss committed:
revert: `stackhpc-ci` back to `stackhpc/xena`

markgoddard commented on .automation.conf/config.sh:
```diff
@@ -28,3 +28,6 @@ fi
 fi
+
+export KAYOBE_AUTOMATION_TEMPEST_LOADLIST=tempest-full
+export KAYOBE_AUTOMATION_TEMPEST_SKIPLIST=tempest-full
```
Let's keep this file as it was.

markgoddard commented on .automation.conf/tempest/tempest-ci-multinode.overrides.conf:
```diff
@@ -0,0 +1,30 @@
+
+[volume]
+min_microversion = 3.0
+max_microversion = 3.64
```
3.66 for xena

markgoddard commented on etc/kayobe/ansible/requirements.yml:
```diff
@@ -1,8 +1,10 @@
 ---
```
No need to change this file.

markgoddard commented on etc/kayobe/environments/ci-multinode/README.md:
```diff
@@ -1,48 +1,233 @@
 # Multinode Test Environment 
```
I moved this file to the new documentation, doc/source/contributor/environments/ci-multinode.rst.

bbezak commented on etc/kayobe/environments/ci-multinode/kolla/config/keepalived/keepalived.conf:
```diff
@@ -0,0 +1,48 @@
+    advert_int 3 # Increase advert interval from 1 to 3 as there was a lot of jumping amongst controllers on the internal network 
+{% if keepalived_traffic_mode == 'unicast' %}
+    unicast_src_ip {{ api_interface_address }}
+{% if groups['haproxy'] | length > 1 %}
```
haproxy group was renamed to load balancer

bbezak commented on etc/kayobe/environments/ci-multinode/kolla/config/keepalived/keepalived.conf:
```diff
@@ -0,0 +1,48 @@
+    unicast_src_ip {{ api_interface_address }}
+{% if groups['haproxy'] | length > 1 %}
+    unicast_peer {
+{% for host in groups['haproxy'] %}
```
same here

markgoddard commented on .automation.conf/config.sh:
```diff
@@ -3,7 +3,7 @@
 # See: https://github.com/stackhpc/docker-rally/blob/master/bin/rally-verify-wrapper.sh for a full list of tempest parameters that can be overriden.
 # You can override tempest parameters like so:
-export TEMPEST_CONCURRENCY=8
+export TEMPEST_CONCURRENCY=6
```
Need to bear in mind this is a shared file that affects all environments.

jackhodgkiss replied:
Could we not use conditional statements to test the `KAYOBE_ENVIRONMENT`? 

if `ci-multinode` 6 else 8.

markgoddard replied:
Looks like the original value was 2

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -457,7 +457,7 @@ kolla_build_args:
 #kolla_enable_ceilometer_horizon_policy_file:
 #kolla_enable_ceilometer_ipmi:
 #kolla_enable_cells:
-kolla_enable_central_logging: true
```
Please don't modify these global flags.

jackhodgkiss replied:
This has been reverted and the disabling of services has been moved into `ci-multinode/kolla.yml`

markgoddard commented on .automation.conf/tempest/load-lists/tempest-full:
```diff
@@ -0,0 +1,1601 @@
+tempest.api.object_storage.test_object_slo.ObjectSloTest.test_upload_manifest[id-2c3f24a6-36e8-4711-9aa2-800ee1fc7b5b]
```
I do wonder how portable this list is between different environments. Probably something to ask about in Slack. i.e. do different environments tend to have different "full" test lists, and if so, should they?

markgoddard replied:
I've asked about this in slack

markgoddard commented on .automation.conf/tempest/skip-lists/tempest-full:
```diff
@@ -0,0 +1 @@
+tempest.scenario.test_network_basic_ops.TestNetworkBasicOps.test_subnet_details.*: "Cirros image doesn't have '/var/run/udhcpc.eth0.pid"
```
The skip list is probably fairly environment-specific. We might need to give it a multinode-specific name, and conditionally use it in config.sh.

markgoddard replied:
Still a problem

markgoddard commented on etc/kayobe/ansible/requirements.yml:
```diff
@@ -1,7 +1,7 @@
 collections:
   - name: stackhpc.cephadm
-    version: 1.10.0
+    version: 1.12.1
```
Is this required for a bug fix or something? Probably deserves a quick release note.

jackhodgkiss replied:
Added small release note to reflect the bump in version.

markgoddard commented:
pep8 is failing

markgoddard commented on etc/kayobe/ansible/requirements.yml:
```diff
@@ -1,7 +1,7 @@
 collections:
   - name: stackhpc.cephadm
-    version: 1.10.0
+    version: 1.12.1
```
I'm still concerned about this. It introduces the `cephadm_host_labels` bug which is going to affect all deployments and is annoying to diagnose. I'd much rather we had a fix in the collection.

markgoddard replied:
Probably the way to fix it is using `| default([])` when referencing the variable in the collection.

jackhodgkiss commented:
> pep8 is failing

Resolved this issue ran locally before pushing.

jackhodgkiss committed:
feat: revert `TEMPEST_CONCURRENCY` to two

jackhodgkiss committed:
feat: update address of `pulp-server`

jackhodgkiss committed:
fix: ensure `cephadm_host_labels` is defined

jackhodgkiss committed:
feat: load `ci-multinode` tempest settings conditionally

Firstly, the skip-list has been renamed from `tempest-full` to `ci-multinode` as this
list would be environment specific. Secondly the `config.sh` script now contains a
conditional for the `ci-multinode` environment allowing for the test list and
concurrency settings to be loaded without impacting other environments.

jackhodgkiss committed:
fix: remove extra blank lines

jackhodgkiss committed:
feat: reduce concurrency and remove load list

jackhodgkiss committed:
fix: remove unused collection

jackhodgkiss committed:
fix: README has been moved

jackhodgkiss committed:
feat: disable some unwanted service within ci-multinode

jackhodgkiss committed:
feat: `haproxy` group has been replaced with `loadbalancer`

jackhodgkiss committed:
feat: increase `max_microversion`

jackhodgkiss committed:
fix: restore order to `requirements.yml`

jackhodgkiss committed:
fix: add missing new line

jackhodgkiss committed:
fix: use `HTTP` dashboard URL over `HTTPS`

jackhodgkiss committed:
feat: bump `stackhpc.cephadm` version

jackhodgkiss committed:
fix: reduce multinode tempest concurrency

markgoddard commented on .automation.conf/config.sh:
```diff
@@ -19,6 +19,11 @@ if [ ! -z ${KAYOBE_ENVIRONMENT:+x} ]; then
     export TEMPEST_CONCURRENCY=1
   fi
+  if [[ "$KAYOBE_ENVIRONMENT" =~ "ci-multinode" ]]; then
+      # Increase concurrency for multinode environment. Eight has proven to high, leading to failing tests due to poor CEPH performance.
```
Comment no longer relevant

jackhodgkiss replied:
Removed irrelevant comment.

markgoddard commented on releasenotes/notes/cephadm-bump-294148e93121da48.yaml:
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    Increase `stackhpc.cephadm` collection to version `1.12.1`.
```
1.12.2

jackhodgkiss replied:
Fixed.

markgoddard commented on releasenotes/notes/bump-cephadm-version-e22b37907acf0efd.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Increase the required version of `stackhpc.cephadm` collection to avoid an
+    issue where deployment of cephadm would fail if `cephadm_host_labels` was 
+    absent from group or host vars.
```
This bug wasn't in the 1.10.0 release, so no need for a reno

jackhodgkiss replied:
Removed.

jackhodgkiss committed:
feat: bump `stackhpc.cephadm` collection version

jackhodgkiss committed:
feat: remove unecessary comment

markgoddard reviewed (approved):
Good work Jack.
